metpy/plots/declarative.py

```diff
@@ -802,7 +804,7 @@ def draw(self):
 class ImagePlot(Plot2D):
     """Represent an image plot."""
 
-    @observe('colormap', 'image_range')
+    @observe('colormap', 'colorbar', 'image_range')
```

Comment:
So here, we're looking at attributes for whom it's easy to change value--change them and do a simple redraw and everything works. For the colorbar, that requires much bigger changes to the figure, so I think we need to rebuild the whole thing. Actually, given that the colorbar carves out space on the figure, one could argue this should be on `Panel`/`MapPanel` or even `PanelContainer`.

For now, let's make a new handler for this one, based on what's in `ContourPlot`:
```python
    @observe('colorbar')
    def _set_need_rebuild(self, _):
	"""Handle changes to attributes that need to regenerate everything."""
	# Because matplotlib doesn't let you just change these properties, we need
        # to trigger a clear and re-call of contour()
        self.clear()
```

Reply:
Okay, I've done this, will be able to refactor later as needed.